app/avo/actions/change_email.rb

```diff
@@ -0,0 +1,25 @@
+class ChangeEmail < BaseAction
+  field :from_email, name: "Email", as: :text, required: true
+
+  self.name = "Change Email"
+  self.visible = lambda {
+    current_user.team_member?("rubygems-org") && view == :show
+  }
+
+  self.confirm_button_label = "Change Email"
+
+  class ActionHandler < ActionHandler
+    set_callback :handle, :before do
+      error "Must enter Email" unless fields[:from_email].presence
+    end
+
+    def handle_model(user)
+      user.email = fields["from_email"]
+      user.email_confirmed = false
+      user.generate_confirmation_token
+
+      return unless user.save
```

Comment:
should this be `.save!` so the validation error gets bubbled up as an alert?

Reply:
Done here Makes sense. Done here https://github.com/rubygems/rubygems.org/pull/3709/commits/4c161ff545946b77fc6952c4296b5fbddaa249c6